Harvey-OS/harvey
# Reintroduce libdisk changes from old Harvey

commit 275f0b42411259f8078477cf24f2b338d47c6ab2
Author: Dan Cross <cross@gajendra.net>
Date:   Thu Apr 20 16:19:04 2017 +0000

    Really expose warnings

    But also make it so that we can continue building.
    By exposing them, we see them so we can fix them.

    Signed-off-by: Dan Cross <cross@gajendra.net>

commit 331ffebb5394eb0768cea498e43a2d2c98829e9d
Author: Sevki Hasirci <s@sevki.org>
Date:   Mon Feb 22 00:55:22 2016 +0100

    just the files

commit 400364d361586e548bd7c08d8cdcb64a4bc69e90
Author: Ronald G. Minnich <rminnich@gmail.com>
Date:   Wed Jul 6 13:53:16 2016 -0700

    fixfmt: with this last round of changes I can boot.

    Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

commit 48090c023db7a9d0f4f36f3b270dc30080c24ce5
Author: ron minnich <rminnich@gmail.com>
Date:   Sun Feb 14 05:49:03 2016 +0100

    Revert "build: added new build files"
    Does not work at all. Can't build a working system with it yet.

    Let's take this slower, ok? Not break it all next time.

    This reverts commit 4fd75744926150212a0c0cd3c5a2ea51d0637d12.

    Change-Id: I91fce92249dd9d7bf3ca731a124c61a499382dd2

commit 493f272fd0c04d30e26b0b410ca588ee86a59c6e
Author: Ronald G. Minnich <rminnich@gmail.com>
Date:   Sun Apr 24 12:40:37 2016 -0700

    Remove plan9 -extensions.

    Do these still work? I don't know. Someone test, please.

    Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

commit 4fd75744926150212a0c0cd3c5a2ea51d0637d12
Author: Sevki <sevki@spotify.com>
Date:   Wed Jan 27 12:46:31 2016 +0100

    build: added new build files

    kernel compiles and crashes

    closes #37

    to test use

            go get -u sevki.org/build/cmd/build
            build //sys/src/9/amd64:harvey
    or

            build -v //sys/src/9/amd64:harvey

    inside a editor like emacs or acme.

    If you are building on OS X you should also have something like .build
    file at the root of the project and add something like

            CC: gcc
            TOOLPREFIX: x86_64-elf-

    Change-Id: Ib6f6156eee1936ceb5f8b0bfa64ed45589195c31
    Signed-off-by: Sevki <sevki@spotify.com>

commit 6533db6114e97d5687bdb6a0f94136c98d009408
Author: Dan Cross <cross@gajendra.net>
Date:   Mon May 1 00:00:16 2017 +0000

    Remove trailing whitespace from source files.

    This is a trivial cleanup (done with a script, Ron!).

    Signed-off-by: Dan Cross <cross@gajendra.net>

commit 7218354a02aa0aa4808a862cd8a08dc16f9edcb9
Author: Ronald G. Minnich <rminnich@gmail.com>
Date:   Tue Dec 8 13:31:08 2015 -0800

    usb: mv scsierrs.c to libdisk

    This is the start of work to make usb more like a real
    system instead of a bag on the side.

    It will require moving .h files, creating a libusb, and separating
    commands from libraries.

    First step is to move things out of usb that are more general.

    scsierrs.c is not only usb, it's everything scsi.

    We can eventually consider vendoring this again from the web site but
    it has not changed in many years. It may not be worth it. It may be simpler
    just to do it by hand on occasion.

    Builds and works fine.

    Change-Id: I15c32c253287fb379459e1691cb9f5326d7ab6ca
    Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

commit ba14fa27776ef2371a42607ec04bf50287fbd0be
Author: Ronald G. Minnich <rminnich@gmail.com>
Date:   Fri Nov 13 08:20:41 2015 -0800

    New json format per Giacomo's ideas.

    Instead of this:
    {
            "Name": "a",
            ...
    }

    Do this:
    {
            "Name": {
                    ...
            }
    }

    This is way better. Now the name is visible at the top level of any json viewer, we can
    have multiple stanzas per file, the name is way more easy to find, ... the advantages are legion.

    This includes a change to build.go but not to preen.

    This now works fine, BUILD all produces a working kernel and userland. But someone else needs to verify it.

    Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

    Change-Id: I9b1cf2597e582895b8f3ab127bc9c1d77de96cbd

commit c9bec0b2ba4272968072bbb58237fda58725088d
Author: Ronald G. Minnich <rminnich@gmail.com>
Date:   Fri Apr 22 16:02:25 2016 -0700

    Move plan9-extensions option into only those places that need it

    Now you can start to fix the individual things. It's not in
    core.json any more.

    You can start with bufio, which is a mess.

    Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

commit e087b93e8188160b096e82e5c9bf27657118847f
Author: Ronald G. Minnich <rminnich@gmail.com>
Date:   Wed Jul 6 13:48:15 2016 -0700

    fixfmt: more fixes. Also, always include ratrace in ramfs

    It makes debugging when booting from rc far, far easier.

    Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

commit eacf1b3e899331f25ee39c2aa438b5f23e538275
Author: Dan Cross <cross@gajendra.net>
Date:   Fri Apr 7 20:47:23 2017 +0000

    Clean up a number of warnings.

    Clean up a number of warnings that pop up when compiling
    with more stringent compiler warnings.

    Signed-off-by: Dan Cross <cross@gajendra.net>

commit f6e68a90f971b8dc021fe4f4c97fa9234de37f17
Author: Ronald G. Minnich <rminnich@gmail.com>
Date:   Thu Jul 7 19:52:40 2016 -0700

    fixfmt: Still MORE problems.

    %x.yu[doX] -> %x.y[uoX]

    Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

Signed-off-by: Graham MacDonald <grahamamacdonald@gmail.com>
